packages/x-date-pickers/src/dateTimeViewRenderers/dateTimeViewRenderers.tsx

```diff
@@ -0,0 +1,181 @@
+import * as React from 'react';
+import Divider from '@mui/material/Divider';
+import { resolveComponentProps } from '@mui/base/utils';
+import { DateCalendar, DateCalendarProps } from '../DateCalendar';
+import { DateOrTimeViewWithMeridiem } from '../internals/models';
+import {
+  MultiSectionDigitalClock,
+  MultiSectionDigitalClockProps,
+  multiSectionDigitalClockSectionClasses,
+} from '../MultiSectionDigitalClock';
+import { DateTimeViewWrapper } from '../internals/components/DateTimeViewWrapper';
+import { isInternalTimeView } from '../internals/utils/time-utils';
+import { isDatePickerView } from '../internals/utils/date-utils';
+import type { DateTimePickerProps } from '../DateTimePicker/DateTimePicker.types';
+
+export interface DateTimeViewRendererProps<TDate>
+  extends Omit<
+      DateCalendarProps<TDate> & MultiSectionDigitalClockProps<TDate>,
+      | 'views'
+      | 'openTo'
+      | 'view'
+      | 'onViewChange'
+      | 'focusedView'
+      | 'components'
+      | 'componentsProps'
+      | 'slots'
+      | 'slotProps'
+    >,
+    Pick<DateTimePickerProps<TDate>, 'components' | 'componentsProps' | 'slots' | 'slotProps'> {
+  view: DateOrTimeViewWithMeridiem;
+  onViewChange?: (view: DateOrTimeViewWithMeridiem) => void;
+  views: readonly DateOrTimeViewWithMeridiem[];
+  focusedView: DateOrTimeViewWithMeridiem | null;
+  timeViewsCount: number;
+}
+
+export const renderDesktopDateTimeView = <TDate extends unknown>({
+  view,
+  onViewChange,
+  views,
+  focusedView,
+  onFocusedViewChange,
+  value,
+  defaultValue,
+  onChange,
+  className,
+  classes,
+  disableFuture,
+  disablePast,
+  minDate,
+  minTime,
+  maxDate,
+  maxTime,
+  shouldDisableDate,
+  shouldDisableMonth,
+  shouldDisableYear,
+  shouldDisableTime,
+  shouldDisableClock,
+  reduceAnimations,
+  minutesStep,
+  ampm,
+  onMonthChange,
+  monthsPerRow,
+  onYearChange,
+  yearsPerRow,
+  defaultCalendarMonth,
+  components,
+  componentsProps,
+  slots,
+  slotProps,
+  loading,
+  renderLoading,
+  disableHighlightToday,
+  readOnly,
+  disabled,
+  showDaysOutsideCurrentMonth,
+  dayOfWeekFormatter,
+  sx,
+  autoFocus,
+  fixedWeekNumber,
+  displayWeekNumber,
+  disableIgnoringDatePartForTimeValidation,
+  timeSteps,
+  skipDisabled,
+  timeViewsCount,
+}: DateTimeViewRendererProps<TDate>) => {
+  const isActionBarVisible = !!resolveComponentProps(
```

Comment:
OK that's super not sexy...
But not sure how to do better

Reply:
Agreed. 🙈 
But given layout, I didn't see other options of how to cleanly avoid having the unwanted divider/border when no actions are present. 🤷